docs/ExpectAPI.md

```diff
@@ -255,6 +255,41 @@ it('stores only 10 characters', () => {
 });
 ```
 
+#### async
+
+If your custom inline snapshot matcher is async i.e. uses `async`-`await` you might encounter an error like "Multiple inline snapshots for the same call are not supported". Jest needs additional context information to find where the custom inline snapshot matcher was used to update the snapshots properly.
+
+```js
+const {toMatchInlineSnapshot} = require('jest-snapshot');
+
+expect.extend({
+  async toMatchObservationInlineSnapshot(fn, ...rest) {
+    // The error (and its stacktrace) must be created before any `await`
```

Comment:
I wonder if Jest itself could instantiate an error and store it before calling this function? So that particular gotcha is gone. Not needed for now of course, and documenting for Jest 25 and 26 is great regardless

Reply:
Yeah, I did some shallow debugging but couldn't really figure out why it was working without `this.error = new Error()` for existing snapshots but not for new snapshots. I might revisit this though having a test in the repo is a sufficient guarantee for me now :wink: